### PR TITLE
chore: update typescript deps and co

### PR DESCRIPTION
**Pull request by skjnldsv**
```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: eslint-config-standard@17.0.0
npm ERR! Found: eslint-plugin-n@16.0.1
npm ERR! node_modules/eslint-plugin-n
npm ERR!   peer eslint-plugin-n@"^16.0.0" from @nextcloud/eslint-config@8.3.0-beta.1
npm ERR!   node_modules/@nextcloud/eslint-config
npm ERR!     dev @nextcloud/eslint-config@"^8.3.0-beta.1" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer eslint-plugin-n@"^15.0.0" from eslint-config-standard@17.0.0
npm ERR! node_modules/eslint-config-standard
npm ERR!   peer eslint-config-standard@"^17.0.0" from @nextcloud/eslint-config@8.3.0-beta.1
npm ERR!   node_modules/@nextcloud/eslint-config
npm ERR!     dev @nextcloud/eslint-config@"^8.3.0-beta.1" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: eslint-plugin-n@15.7.0
npm ERR! node_modules/eslint-plugin-n
npm ERR!   peer eslint-plugin-n@"^15.0.0" from eslint-config-standard@17.0.0
npm ERR!   node_modules/eslint-config-standard
npm ERR!     peer eslint-config-standard@"^17.0.0" from @nextcloud/eslint-config@8.3.0-beta.1
npm ERR!     node_modules/@nextcloud/eslint-config
npm ERR!       dev @nextcloud/eslint-config@"^8.3.0-beta.1" from the root project

```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,13 @@
 
 ## Unreleased
 
+## [v8.3.0-beta.1](https://github.com/nextcloud/eslint-config/tree/v8.3.0-beta.2) (2023-06-27)
```
Comment (susnux) [inline CHANGELOG.md:5]:
```suggestion
## [v8.3.0-beta.2](https://github.com/nextcloud/eslint-config/tree/v8.3.0-beta.2) (2023-06-27)
```

File: package-lock.json [hunk 2]
```diff
@@ -1,31 +1,31 @@
 {
   "name": "@nextcloud/eslint-config",
-  "version": "8.3.0-beta.1",
+  "version": "8.3.0-beta.2",
   "lockfileVersion": 2,
```
Comment (susnux) [inline package-lock.json:4]: Maybe do a `npm update` or remove the file and create a fresh one with `npm i` to update the lockfile to version 3 (much smaller by dropping NPM < 7 support, v3 just supports NPM 7+ and is the default for our minimum of NPM 9).

**Comment by skjnldsv** [comment]
Done!

**Comment by susnux** [comment]
While you are at it maybe run `npm audit fix` at least for me there are 30 issues before and only 27 after :see_no_evil:

**Review by susnux — APPROVED** [review]
:sparkles:

**Pull request merged**